packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -454,7 +455,8 @@ class ScrollableState<T extends Scrollable> extends State<T> {
   /// If there are no in-progress scrolling physics, this function scrolls to
   /// the given offset instead.
   void didUpdateScrollBehavior(double newScrollOffset) {
-    setState(() { /* The scroll behavior is part of our build state. */ });
+    // This does not call setState, because if anything below actually
```

Comment:
If the scroll behavior's contentExtent changes, but the scrollOffset does not, and there aren't scrolls in progress, I think we still need to redraw.